astreamer/audio_stream.cpp

```diff
@@ -1071,11 +1076,19 @@ void Audio_Stream::closeAudioQueue()
     delete m_audioQueue, m_audioQueue = 0;
 }
     
+UInt64 Audio_Stream::defaultContentLength()
+{
+    return m_defaultContentLength;
+}
+    
 UInt64 Audio_Stream::contentLength()
 {
     if (m_contentLength == 0) {
         if (m_inputStream) {
             m_contentLength = m_inputStream->contentLength();
+            if (m_contentLength == 0) {
```

Comment:
I suspect this isn't the correct place to do the contentLength -> defaultContentLength override.

Do you think I should pass the defaultContentLength into ```http_stream``` and then simply set contentLength to the defaultContentLength in the case of no content-length header?